### ISSUE TITLE
Exception error implementing Sphinx's tutorial Developing a "todo" extension

### ISSUE THREAD

**Issue by pique-b** [sphinx-doc/sphinx#11826]
### Describe the bug

I am trying to follow the captioned tutorial at [https://www.sphinx-doc.org/en/master/development/tutorials/todo.html](https://www.sphinx-doc.org/en/master/development/tutorials/todo.html).

However, when I am trying to build `html` document, I get an exception saying:

`Extension error:`
`Handler <function process_todo_nodes at 0x106891ae8> for event 'doctree-resolved' threw an exception`
`make: *** [html] Error 2`

When I change the extension name from `todo` to `sphinx.ext.todo` like shown below:

`extensions = ['`
` ...`
`'sphinx.ext.todo'` # This works
`# 'todo'` # This raises **exception**
`...`
`]`

the directive `todo` when added to the source docs (`.rst`) **works as expected** (when used with `sphinx.ext.todo`).

### How to Reproduce

I have completed all the steps described in the tutorial:

- Created folder `_ext` in folder `source`
- Created file `todo.py` in folder `_ext`
- Copy / pasted the codes described in [https://www.sphinx-doc.org/en/master/development/tutorials/todo.html#writing-the-extension](https://www.sphinx-doc.org/en/master/development/tutorials/todo.html#writing-the-extension)
- Usage: Followed the instructions in [https://www.sphinx-doc.org/en/master/development/tutorials/todo.html#using-the-extension](https://www.sphinx-doc.org/en/master/development/tutorials/todo.html#using-the-extension):
- In `conf.py` added `_ext` directory to python path like so:
  `import os` # Original line
   `import sys` # Original line

   `sys.path.insert(0, os.path.abspath('.'))` # Original line
   `sys.path.append(os.path.abspath('./_ext'))` # Added the path to _ext folder
- Added the newly created extension `todo` like so:
   `extensions = ['sphinx.ext.autodoc',`
     `...`
     `'todo',`
    `]`
- Added the following as well:
   `todo_include_todos = False`

### Environment Information

```text
Python : 3.6
Sphinx : 3.1.2
MacOs : Sierra 10.12.6

Currently using the following extensions:

extensions = ['sphinx.ext.autodoc',
              'sphinx.ext.imgmath',
              'sphinx.ext.doctest',
              'sphinx.ext.coverage',
              'sphinx.ext.intersphinx',
              'sphinx.ext.extlinks',
              'sphinx_togglebutton',
              'sphinx.ext.todo',
              #'todo',
]
```

### Additional context

1. In order to check, if Sphinx is accessing the newly created python file `todo.py`, carried out some changes in the script in the file (`todo.py`). But there was no change in the behavior during **_build_**, as if the file `todo.py` does not exist.
2. I have added a blank `__ini__.py` to the folder `_ext` also, to see if that made any difference, but **it did not**.

**Comment by picnixz** [comment]
Hi, first of all we don't support older sphinx versions nor do we support Python 3.6 (only 3.9+ currently).

On the other hand the tutorial might have been out of date and we need to update it if needed. I cannot check it now so I would appreciate if someone can reproduce the issue with the latest Sphinx version. 

Can you check if the tutorial for the hello world extension works? Or can you just debug the function raising the exception? there might have been some  bug inside that we left.

**Comment by pique-b** [comment]
Hi,

Thanks for the reply.

Yes, tried the **helloworld** tutorial and it is working perfectly.

So there is certainly something wrong with the script in the `todo.py` file. I am trying to debug as best as I can. I will post in case there is any development.

In the meantime, may I request you to kindly check at your end as well.

Thanks again.

**Comment by electric-coder** [comment]
@pique-b can you please fix the post so the links point to the right URLs please? All 3 links points to the same wrong address...

**Comment by pique-b** [comment]
@electric-coder I am not sure why it's happening, but all **three** links are from the same page currently open on my browser. It is from the same page that I tried the codes in question.

Funny though, when I am putting the same URL in my reply to you and checking it (from the **Preview** pane), it's working perfectly.

https://www.sphinx-doc.org/en/master/development/tutorials/todo.html

And you are right, the link in my original post takes one to a page with **No results matched your search.** messsage!!

**Comment by picnixz** [comment]
@pique-b I fixed your links (you actually forgot the URL in your markdown).

I tried locally with the latest Sphinx and Python 3.9 and 3.11 and it works. So I think it's because of your old version and / or your Python version. Unfortunately, we only support the latest Sphinx version (there is no backports). The reason is that it's too much work for backporting everything (especially because of how the AST of major Python versions can change).
